src/main/java/functions/Parser.java

```diff
@@ -0,0 +1,139 @@
+package functions;
+
+import commands.*;
+import tasks.Task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Utility class for parsing user input and generating appropriate commands.
+ */
+public class Parser {
+
+    /**
+     * Enumeration representing different date and time formats for parsing.
+     */
+    enum DateTimeFormat {
+        FORMAT1("yyyy-MM-dd HH:mm"),
+        FORMAT2("dd.MM.yyyy HH:mm"),
+        FORMAT3("MM/dd/yyyy HH:mm"),
+        FORMAT4("yyyy-MM-dd hh:mm a"),
+        FORMAT5("dd.MM.yyyy hh:mm a"),
+        FORMAT6("MM/dd/yyyy hh:mm a"),
+        FORMAT7("yyyy-MM-dd HHmm"),
+        FORMAT8("dd.MM.yyyy HHmm"),
+        FORMAT9("MM/dd/yyyy HHmm");
+
+        private final DateTimeFormatter formatter;
+
+        DateTimeFormat (String pattern) {
+            this.formatter = DateTimeFormatter.ofPattern(pattern);
+        }
+
+        public DateTimeFormatter getFormatter() {
+            return formatter;
+        }
+    }
+
+    /**
+     * Parses a date string using various date and time formats.
+     *
+     * @param dateString The date string to be parsed.
+     * @return A LocalDateTime object if parsing is successful, otherwise null.
+     */
+    public static LocalDateTime parseDate(String dateString) {
+        for (DateTimeFormat format : DateTimeFormat.values()) {
+            try {
+                LocalDateTime d = LocalDateTime.parse(dateString, format.getFormatter());
+                return d;
+            } catch (DateTimeParseException e) {
+                // Try the next format
+            }
+        }
+        // None of the formats matched
+        return null;
+    }
+
+    /**
+     * Parses the user input and generates the corresponding command object.
+     *
+     * @param input The user input string.
+     * @return A Command object representing the parsed command.
+     */
+    public static Command parse(String input) {
+        if (input.equals("bye")) {
+            return new ExitCommand();
+        }
+        if (input.equals("list")) {
+            return new ListCommand();
+        } else {
+            String[] words = input.split(" ");
+            String first = words[0];
+            if (first.equals("delete")) {
+                int s = Integer.valueOf(words[1]);
+                return new DeleteCommand(s);
+            } else if (first.equals("mark")) {
+                int s = Integer.valueOf(words[1]);
+                return new MarkCommand(s);
+            } else if (first.equals("unmark")) {
+                int s = Integer.valueOf(words[1]);
+                return new UnmarkCommand(s);
```

Comment:
Perhaps you could put the result directly into the arguments? For example:
```suggestion
                return new UnmarkCommand(Integer.valueOf(words[1]));
```
I'm not so sure about the usage of `s` :confused: